ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -1010,38 +1041,37 @@ public void write() throws StoreException {
       try {
         if (indexes.size() > 0) {
           // before iterating the map, get the current file end pointer
-          Map.Entry<Long, IndexSegment> lastEntry = indexes.lastEntry();
+          Map.Entry<Offset, IndexSegment> lastEntry = indexes.lastEntry();
```

Comment:
With compaction, did you say you are going to embed `indexes` (or rather `segments` as this should be named) inside an `AtomicReference`? Because otherwise, this persistence is going to be tricky and you would have to make use of an iterator to traverse.

In fact, this code should really have been using an iterator over the segments rather than going with this `lowerEntry` business. I understand that you don't want to make extensive changes now so it is fine - but you might want to revisit this during compaction.